mikrotik/resource_pool.go

```diff
@@ -0,0 +1,103 @@
+package mikrotik
+
+import (
+	"github.com/ddelnano/terraform-provider-mikrotik/client"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourcePool() *schema.Resource {
+	return &schema.Resource{
+		Create: resourcePoolCreate,
+		Read:   resourcePoolRead,
+		Update: resourcePoolUpdate,
+		Delete: resourcePoolDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"ranges": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"comment": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourcePoolCreate(d *schema.ResourceData, m interface{}) error {
+	name := d.Get("name").(string)
+	ranges := d.Get("ranges").(string)
+	comment := d.Get("comment").(string)
+
+	c := m.(client.Mikrotik)
+
+	pool, err := c.AddPool(name, ranges, comment)
+	if err != nil {
+		return err
+	}
+
+	return poolToData(pool, d)
+}
+
+func resourcePoolRead(d *schema.ResourceData, m interface{}) error {
+	c := m.(client.Mikrotik)
+
+	pool, err := c.FindPool(d.Id())
+
+	if err != nil {
+		return err
+	}
+
```

Comment:
I'm pretty confident this code won't work for the following scenario:
1. User creates pool with the provider
2. User deletes pool in the mikrotik UI
3. User runs a terraform plan expecting terraform to recreate the pool but instead sees an error

This is because the `NotFound` error will be returned from `FindPool`. We need to explicitly check the error type and handle it accordingly.

Please write a test for this (see the `mikrotik_dns_record` resource tests for an example on how to do this)